components/layout/model.rs

```diff
@@ -451,12 +451,18 @@ pub fn style_length(style_length: LengthOrPercentageOrAuto,
     }
 }
 
+pub fn maybe_auto(au: Option<Au>) -> MaybeAuto {
+    match au {
+        Some(l) => MaybeAuto::Specified(l),
+        None    => MaybeAuto::Auto,
+    }
+}
+
```

Comment:
after some consideration I suggest to make this a method of `MaybeAuto`, with name like `from_option`, to make it more semantic.